locale/en/docs/guides/event-loop-timers-and-nexttick.md

```diff
@@ -303,6 +306,56 @@ situations because **it allows you to "starve" your I/O by making
 recursive `process.nextTick()` calls**, which prevents the event loop
 from reaching the **poll** phase.
 
+#### Deduplication
+
+For the `timers` and `check` phases, the there is a *single transition 
+between C to JS for multiple immediates and timers*. This deduplication
+is a form of optimzation, which may produce some unexpected side effects.
+Take this code snippet as an example:
+
+```js
+// dedup.js
+const foo = [1, 2];
+const bar = ['a', 'b'];
+
+foo.forEach( num => {
+  setImmediate(() => {
+    console.log('setImmediate', num);
+    bar.forEach(char => {
+      process.nextTick(() => {
+        console.log('process.nextTick', char);
+      });
+    });
+  });
+});
+```
+```bash
+$ node dedup.js
+setImmediate 1
+setImmediate 2
+process.nextTick a
+process.nextTick b
+process.nextTick a
+process.nextTick b
+```
+
+The main thread adds two `setImmediate()` events, which when processed
+will add two `process.nextTick()` events. When the event loop reaches
+the `check` phase, it sees that there are currently two events created by
+`setImmediate()`. The first event is grabbed and processed, which prints
+and adds two events to the `nextTickQueue`.
+
+Because of deduplication, the event loop does not transition back to the
+C/C++ layer to check if there are items in the `nextTickQueue` immediately. It
+instead continues to process any remaining `setImmediate()` events, of which
+one currently remains. After processing this event, two more events are 
+added to the `nextTickQueue` for a total of four events.
+
+At this point, all previously added `setImmediate()` events have been processed.
+The `nextTickQueue` is now checked, and events are processed in FIFO order. When
+these `nextTickQueue` is emptied, the event loop is considers all operations to have 
```

Comment:
these -> this
is considers -> considers
to have completed -> to have been completed?